test/e2e/specs/editor/editor__navbar.ts

```diff
@@ -43,6 +43,12 @@ describe( DataHelper.createSuiteTitle( `Editor: Navbar` ), function () {
 	} );
 
 	it( 'Return to Calypso dashboard', async function () {
-		await editorPage.exitEditor();
+		const isMobileClassicView =
+			envVariables.VIEWPORT_NAME === 'mobile' && ( await page.isVisible( '#wpadminbar' ) );
```

Comment:
**issue:blocking**: I would not use `page.isVisible()` to reliably check whether an element is actually visible. The [official `API](https://playwright.dev/docs/api/class-page#page-is-visible)` isn't too clear on this but [this snippet](https://playwright.dev/docs/api/class-page#page-is-visible-option-timeout), combined with my past experience using `page.isXYZ` methods suggests that it should not be used.

Can I suggest the alternative `locator.isVisible` instead?

Reply:
Thanks for the review, @worldomonation!

> Can I suggest the alternative locator.isVisible instead?

Sure, `locator` sounds good! Do you recall what the specific issue with `page.isVisible` was, though? Seems like both `page.isVisible` and `locator.isVisible` have the `timeout` option marked as `DEPRECATED`. 🤔  I'm wondering if there's any difference between those two implementation-wise since the logic seems identical.

Reply:
> Seems like both page.isVisible and locator.isVisible have the timeout option marked as DEPRECATED

Hmm, that is a little unexpected. I recall the Playwright team mentioning somewhere public that the `page.isXYZ` methods are no longer considered good use and to use the Locator version instead, but the suggestion may have been actually to use `locator.waitFor` instead.

I went searching for information on `page.isXYZ` methods and whether it's good practice. I found https://github.com/microsoft/playwright/issues/12672 which in turn references https://github.com/microsoft/playwright/issues/12224 as to why the`timeout` field was deprecated, but doesn't make statement on whether it's good practice to use these methods. There's also a recent question https://github.com/microsoft/playwright/issues/19396.

I suppose I should amend my suggestion to say `locator.waitFor` instead. Thanks for pointing it out.